libs/net/services/Runners/BaseService.cs

```diff
@@ -191,13 +192,33 @@ public async Task<int> RunAsync()
 
     /// <summary>
     /// Run the service.
+    /// Keep restarting after failures if configured to do so.
     /// </summary>
     /// <returns></returns>
     private async Task RunServiceAsync()
     {
-        using var scope = this.App.Services.CreateScope();
-        var service = scope.ServiceProvider.GetRequiredService<IServiceManager>();
-        await service.RunAsync();
+        var options = this.App.Services.GetRequiredService<IOptions<ServiceOptions>>().Value;
+        do
+        {
+            try
+            {
+                using var scope = this.App.Services.CreateScope();
```

Comment:
Keep looping if configured to do so, but with a delay.  Since it creates a new instance it will have updated state.  I should change all the services so that they don't need to have this logic within them anymore.  Another day...